curvefs/src/client/dentry_cache_manager.cpp

```diff
@@ -46,13 +48,70 @@ using curve::common::WriteLockGuard;
 using NameLockGuard = ::curve::common::GenericNameLockGuard<Mutex>;
 using ::curvefs::client::filesystem::ToFSError;
 
+MetaStatusCode DentryCacheManagerImpl::CheckTxStatus(
+    const std::string primaryKey, uint64_t startTs, uint64_t curTimestamp) {
+    Key4Dentry key4Dentry;
+    if (!key4Dentry.ParseFromString(primaryKey)) {
+        LOG(ERROR) << "CheckTxStatus parse primary key failed, primaryKey = "
+                   << primaryKey;
+        return MetaStatusCode::PARSE_FROM_STRING_FAILED;
+    }
+    return metaClient_->CheckTxStatus(key4Dentry.fsId, key4Dentry.parentInodeId,
+        primaryKey, startTs, curTimestamp);
+}
+
+MetaStatusCode DentryCacheManagerImpl::ResolveTxLock(const Dentry &dentry,
+    uint64_t startTs, uint64_t commitTs) {
+    return metaClient_->ResolveTxLock(dentry, startTs, commitTs);
+}
+
+MetaStatusCode DentryCacheManagerImpl::CheckAndResolveTx(const Dentry& dentry,
+    const TxLock& txLock, uint64_t timestamp, uint64_t commitTs) {
+    auto rt = CheckTxStatus(txLock.primarykey(), txLock.startts(), timestamp);
+    switch (rt) {
+        case MetaStatusCode::TX_COMMITTED:
+            return ResolveTxLock(dentry, txLock.startts(), commitTs);
+        case MetaStatusCode::TX_ROLLBACKED:
+        case MetaStatusCode::TX_TIMEOUT:
```

Comment:
Will server side return `MetaStatusCode::TX_INPROGRESS`?

Reply:
If `MetaStatusCode::TX_INPROGRESS` returned, the rpc will retry until the states changed to others